### DISCUSSION TITLE
Hide search field and navigate with hjkl

### DISCUSSION THREAD

**Discussion by sohanemon** [ibhagwan/fzf-lua#1618] [Q&A] [answered]
How can I naviage up/down with j/k and also hide the search bar from here?

![image](https://github.com/user-attachments/assets/23306e86-e690-4406-bb06-d3cec053099d)

```lua
buffers = {
          prompt = "   ",
          no_header = true,
          fzf_opts = { ["--delimiter"] = " ", ["--with-nth"] = "-1.." },
          winopts = {
            title = " Switch Buffer ",
            preview = { hidden = "hidden" },
            title_pos = "center",
            width = 0.60,
            height = 0.80,
          },
        }
  ````

**Comment by sohanemon** [comment]
i have added this for navigate:
```lua
keymap = {
        fzf = {
          ["h"] = "abort",
          ["j"] = "down",
          ["k"] = "up",
          ["l"] = "accept",
        },
      }
```

But, is there any way to hide the search bar?

**Reply by ibhagwan** [reply]
> i have added this for navigate:

Note that you aren’t setting the other default keybinds, if this isn’t intended add true to the first index:
```lua

keymap = {
        fzf = {
          true,
          ["h"] = "abort",
          ["j"] = "down",
          …
```

> But, is there any way to hide the search bar

I’m not sure what the “search bar” is the prompt block? Mark it in the screenshot if possible.

**Reply by DaniBedz** [reply]
> i have added this for navigate:
> 
> ```lua
> keymap = {
>         fzf = {
>           ["h"] = "abort",
>           ["j"] = "down",
>           ["k"] = "up",
>           ["l"] = "accept",
>         },
>       }
> ```
> 
> But, is there any way to hide the search bar?

How do you search for anything that contains those characters?

**Reply by ibhagwan** [reply]
> How do you search for anything that contains those characters?

You can’t :)

**Reply by sohanemon** [reply]
> I’m not sure what the “search bar” is the prompt block? Mark it in the screenshot if possible.

yes, the prompt block. I want to hide it.
![image](https://github.com/user-attachments/assets/242763a1-a363-4ca0-a1a0-224e7ea2482c)

**Reply by sohanemon** [reply]
> How do you search for anything that contains those characters?

I don't want to search in buffers. I just want to navigate

**Reply by sohanemon** [reply]
> Note that you aren’t setting the other default keybinds, if this isn’t intended add true to the first index:
> 
> ```lua
> keymap = {
>         fzf = {
>           true,
>           ["h"] = "abort",
>           ["j"] = "down",
>           …
> ```

Im getting this error, setting true at the first index.

Error executing Lua callback: ...man/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/utils.lua:492: attempt to index upvalue 'k' (a number value)

**Reply by ibhagwan** [reply]
> > I’m not sure what the “search bar” is the prompt block? Mark it in the screenshot if possible.
> 
> yes, the prompt block. I want to hide it.

That’s controlled by neovim’s `guicursor` (as fzf is a terminal buffer), you’re going to need the latest nightly:
```lua
:lua vim.opt.guicursor = "n-v-c-sm:block,i-ci-ve:ver25,r-cr-o:hor20,t:ver25"
```

Might also be possible with 0.10.2 (doesn't have `t` mode), see `:help guicursor`.

**Reply by ibhagwan** [reply]
> Im getting this error, setting true at the first index.
> 
> Error executing Lua callback: ...man/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/utils.lua:492: attempt to index upvalue 'k' (a number value)

Seems something is messed up in your config, just used this with no issue:
```lua
require('fzf-lua').setup({
  keymap = {
    fzf = {
      true,
      ["h"] = "abort",
      ["j"] = "down",
      ["k"] = "up",
      ["l"] = "accept",
    },
  }
})
```

**Reply by ibhagwan** [reply, marked as answer]
I think I understand what you want now, to hide the entire search input line, not just the cursor.

A bit hardly but possible https://github.com/junegunn/fzf/issues/2890#issuecomment-1447115710

**Reply by sohanemon** [reply]
> ```lua
> require('fzf-lua').setup({
>   keymap = {
>     fzf = {
>       true,
>       ["h"] = "abort",
>       ["j"] = "down",
>       ["k"] = "up",
>       ["l"] = "accept",
>     },
>   }
> })
> ```

I tried like this, 

 ```lua
 require('fzf-lua').setup({
  buffers = {
   keymap = {
     fzf = {
       true,
       ["h"] = "abort",
       ["j"] = "down",
       ["k"] = "up",
       ["l"] = "accept",
     },
   }
  }
 })
 ```

**Reply by ibhagwan** [reply]
> I tried like this,

Remove the true in this case as you’re setting it under `buffers` as pickers already inherit global keymap, the true works only at the top level `keymap`.

**Reply by dprophete** [reply]
out of curiosity @ibhagwan , what does `true` do for the top level keymap ?

**Reply by ibhagwan** [reply]
> out of curiosity @ibhagwan , what does `true` do for the top level keymap ?

Inherit all the other default keymap.fzf entries, without it you’re losing on the defaults, if you want to redefine the entire bind table completely you can remove it.

**Comment by ibhagwan** [comment]
From https://github.com/ibhagwan/fzf-lua/discussions/1653#discussioncomment-11698879

FYI, after the latest commit https://github.com/ibhagwan/fzf-lua/commit/12ac915c26ec6337ff96be0b8b2311736cca4c13 you can do the below and keep your `esc` as double purpose to abort from "normal" mode, this also changes the prompt info (e.g. "Files") to be displayed in the title since we lose the prompt:
```lua
require('fzf-lua').setup({
  "default-title", -- Important as we lose the prompt
  keymap = {
    fzf = {
      true,
      ["h"] = "abort",
      ["j"] = "down",
      ["k"] = "up",
      ["l"] = "accept",
      ["i"] = "unbind(h)+unbind(j)+unbind(k)+unbind(l)+unbind(i)+change-prompt(> )",
      ["esc"] = 'transform:case "$FZF_PROMPT" in\n'
          .. '*NORMAL*) echo abort;;\n'
          .. '*) echo "change-prompt(NORMAL - PRESS \'I\' FOR INSERT MODE)'
          .. '+rebind(h)+rebind(j)+rebind(k)+rebind(l)+rebind(i)";;'
          .. '\nesac',
      ["start"] = "change-prompt(NORMAL - PRESS 'I' FOR INSERT MODE)"
    },
  }
})
